spec/unit/transaction/resource_harness_spec.rb

```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env ruby
 require 'spec_helper'
 
-require 'puppet/transaction/resource_harness'
```

Comment:
But I'm also adding a `require 'puppet/type/component'` on the next line which in turn `require 'puppet/transaction'`. If someone fiddles with that chain of requirements, then all tests in this file will break.

We usually try to avoid redundant `require` declarations. Is there any specific reason you don't want that done here?